exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/CryptoFunctions.java

```diff
@@ -0,0 +1,345 @@
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+
+      String sha512 = org.apache.commons.codec.digest.DigestUtils.sha512Hex(input);
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = sha512.getBytes().length;
+      buffer.setBytes(0, sha512.getBytes());
+    }
+
+  }
+
+  @FunctionTemplate(name = "aes_encrypt", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class AESEncryptFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Param
+    VarCharHolder rawKey;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Workspace
+    String key;
+
+    @Workspace
+    SecretKeySpec secretKey;
+
+    @Workspace
+    byte[] keyByteArray;
+
+    @Workspace
+    Cipher cipher;
+
+    @Override
+    public void setup() {
+      key = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawKey.start, rawKey.end, rawKey.buffer);
+      java.security.MessageDigest sha = null;
+      try {
+        keyByteArray = key.getBytes("UTF-8");
+        sha = java.security.MessageDigest.getInstance("SHA-1");
+        keyByteArray = sha.digest(keyByteArray);
+        keyByteArray = java.util.Arrays.copyOf(keyByteArray, 16);
+        secretKey = new SecretKeySpec(keyByteArray, "AES");
+
+        cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+        cipher.init(Cipher.ENCRYPT_MODE, secretKey);
+      } catch (Exception e) {
+        logger.debug(e.getMessage());
+      }
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+      String encryptedText = "";
+      try {
+        encryptedText = javax.xml.bind.DatatypeConverter.printBase64Binary(cipher.doFinal(input.getBytes("UTF-8")));
+      } catch (Exception e) {
+        logger.debug(e.getMessage());
+      }
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = encryptedText.getBytes().length;
+      buffer.setBytes(0, encryptedText.getBytes());
+    }
+
+  }
+
+
+  @FunctionTemplate(name = "aes_decrypt", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class AESDecryptFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Param
+    VarCharHolder rawKey;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Workspace
+    String key;
+
+    @Workspace
+    SecretKeySpec secretKey;
+
+    @Workspace
+    byte[] keyByteArray;
```

Comment:
Fixed in latest version.  I was tweaking the code (and it was late) and originally all that was done in the `eval()` function.